crates/uv/tests/it/pip_install.rs

```diff
@@ -8257,3 +8257,161 @@ fn cyclic_build_dependency() {
     "###
     );
 }
+
+#[test]
+fn install_group() -> Result<()> {
```

Comment:
What happens if multiple `pyproject.toml` files are provided? What happens if a group is present in one but not the other? If a group is present in both?

Reply:
Similarly, what about `uv pip install <dir> --group <name>`?

Reply:
I believe as currently implemented we:
* don't complain if we fail to find a group in the toml
* would independently match and include groups from every toml

This falls out of the implementation basically being "for each sourcefile, for each group in that sourcefile, if the selector built from cli flags matches the name of this group, append that group to the requirements".

not sure about the dir usecase

Reply:
I think we should fail (or at the very least, warn) if we cannot find a group.

I don't mind including groups from every toml, but we should have test coverage.

Reply:
We fail in `uv sync` fwiw

```
❯ uv sync --group test
error: Group `test` is not defined in the project's `dependency-group` table
```

Reply:
Added warnings.